Add enhancement - IBM Cloud provider for Power Virtual Server platform

jaypoulz opened the pull request:
The IBM Cloud provides access to power hardware through their Power Virtual Server offering (Power VS). You can find out more at https://www.ibm.com/cloud/power-virtual-server.

Prashanth684 commented on enhancements/installer/powervs-ipi.md:
```diff
@@ -0,0 +1,237 @@
+
+## Summary
+
+This document describes how [PowerVS][powervs-website] becomes an infra provider for OpenShift. PowerVS is a cloud offering by IBM that aims to provid customers interested in running secure and flexible workloads on IBM Power (ppc64le) hardware without having to develop and maintain an on premise datacenter. The goal of this enhancement would be to leverage the PowerVS APIs to deploy an OpenShift ...
```
I've been talking to @clnperez and others and i think we need to make a distinction between the cloud provider and the platform and make sure that we spell out that IBM Cloud is the provider and the platform is PowerVS. In supporting PowerVS as a platform we might have to implement certain parts of the IBM cloud provider that we need to make sure we explicitly call out.

jaypoulz replied:
I believe this newest version does a better job of calling that out explicitly. Please review and resolve if this is addressed to your liking.

jaypoulz replied:
Going to mark this as resolved, and someone can reopen if I missed something important. :)

clnperez commented:
Thank you SO MUCH @jaypoulz for kicking this off! 

Not at all important for technical discussions, but we've been asked to use `Power VS` (note the space) or `Power Virtual Servers`. I promise: This is the only time I'll be that person who nags us about naming. :laughing:

manojnkumar commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
In addition to flavors, there are other aspects of provisioning compute:

Dedicated vs Shared
Class of server (s922 vs e980)

And provisioning storage
Tier1 or Tier3

Should these be baked into the installer or do these controls get exposed?

Prashanth684 replied:
@manojnkumar is there a high level install type that can achieve this? for example aws has instance types (like m4, m5) which is essentially maps to a list of resources in a particular tier. the installer can default to a type and it can also expose an option to override this if necessary.

clnperez replied:
@manojnkumar since these are currently asked when creating a Power VS VM Instance, and also since they affect cost,  My thinking was to expose memory, cores, system type, etc., in the platform details of the install config. 
@Prashanth684 there's not anything like that to my knowledge.

jaypoulz replied:
Added a section in open questions to discuss this topic further.

bpradipt replied:
@Prashanth684 no currently it's not there.

manojnkumar reviewed (commented):
Minor suggestions

bpradipt commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
```diff
@@ -0,0 +1,231 @@
+---
+title: ibm-cloud-ipi
```
should the title be changed to `powervs-ipi` to avoid ambiguity ?

Prashanth684 replied:
or `ibm-cloud-powervs-ipi`

jaypoulz replied:
I went with ibm-cloud-for-power-vs-ipi in the latest revision, since it mirrors the updated title of the doc & PR.

bpradipt commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
nit: "PowerVS cloud the OVA format" -> "PowerVS cloud in the OVA format"

bpradipt commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
The cloud store example will be IBM Cloud Object Storage (COS) bucket. That's the only place from where it can be imported

bpradipt commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
We can mention about using IBM Cloud VPC Loadbalancer using the Direct Link offering

bpradipt commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
DirectLink connectivity is free, however VPC load balancer is costed offering. So it might be better to clarify

jaypoulz replied:
Ah - OK. I will rephrase this. For the VPC load balancer, is the additional cost incurred something we should raise to the business side?

Prashanth684 commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
probably this too : https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/299 . as i understand PowerVS support is still not present in the api provider.

jaypoulz replied:
@clnperez @mkumatag @Prashanth684 - My understanding was that the cluster-api-provider was used for adding the computes, and the machine-api-provider for the bootstrap & masters.

Is this correct?

Prashanth684 replied:
the masters and bootstrap are created by the terraform provider through the openshift installer itself. The cluster-api-provider is used to create compute nodes which are managed by the machine-api. the masters are not managed by machine-apis, only the computes. there is no "machine-api-provider" i think. it's just the cluster-api-providers which are used by the machine-api-operator (which itself is a tweaked implementation of cluster-api) to manage the machinesets.

jaypoulz replied:
OK that makes sense. I stole @clnperez's wording below and called it out as "machine-api controllers".

Can y'all double check my wording on the latest revision?

Prashanth684 commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
probably would be good to call out that a terraform-provider would be used for provisioning control plane and the cluster-api-provider for the computes.

Prashanth684 commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
do we want to break this down into the detail that the ova would be first copied from the cloud-object-storage to the PowerVS service instance, a bootimage would be created and then the VM  would be provisioned ?

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Can we put an asterisk or footnote on this one since we haven't finalized the boot image flow? E.g. we might be able to take advantage of a pre-created boot image.

jaypoulz replied:
I linked back to the OVA considerations section and added a superscript asterisk.

clnperez reviewed (commented):
Coming along nicely! Added some comments for review.

jaypoulz commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Pulled out these steps and added corresponding link.
The only thing I wasn't sure about was `openshift/[cluster-|machine-] `
I'm not familiar with this formatting, so I tried making it more regexy with:
`openshift/[cluster-*machine-*]`

If the former makes more sense, LMK and I'll go back to the way you wrote it. :)

jaypoulz commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Added link in latest revision.

candita replied:
At the moment, I heard that the IBM Cloud installer uses CIS (instead of IBM Cloud DNS) because it is a pre-requisite.  They will start working on IBM Cloud DNS next.  CIS is for "public DNS" and IBM Cloud DNS is for "private DNS".    Can you address whether Power VS will also have CIS, or are both of these covered via association with the IBM Cloud API?

clnperez replied:
We've got CIS DNS built in to our code-base atm. Thanks @candita for that comment. We should make this clear here.

clnperez replied:
@jaypoulz here are a few links: https://cloud.ibm.com/docs/cis?topic=cis-about-ibm-cloud-internet-services-cis#dns-features, https://cloud.ibm.com/docs/cis?topic=cis-about-ibm-cloud-internet-services-cis.

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
nit: account instead of environment

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
it might be good to clarify that it's a manual process owned by Power VS (vs by a user)

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
instead of "allow the installer," can we rephrase to "have the installer mirror the current instance-creation requirement for Power VS, which is to specify each of these values explicitly." ? (I just want to make it clear that right now we don't have these flavors in Power VS, and that users specify these values)

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
This makes it sound like it will take three releases to get to GA. Is that the intent?

jaypoulz replied:
Not necessarily, but this is the most likely path. What I'd like to see us do is enumerate the criteria for maturation, and then see how far we get release by release.

clnperez reviewed (commented):
Great updates! A couple more questions and suggestions :D

manojnkumar commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Not understanding what the power-go-client is.  Isn't it `openshift-install destroy cluster`?

clnperez replied:
@manojnkumar that is the command used, yes. the go client will be vendored in

manojnkumar commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Should we cover a per PR CI job, in the Test Plan or Graduation Criteria section or leave it completely independent?

jaypoulz replied:
Added some more specific listings to the Criteria.

manojnkumar commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Improve deployment time?

manojnkumar commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Should we replace KVM via libvirt, with PowerVM/PowerVC?

jaypoulz replied:
Made a note about PowerVM + VC as well.

manojnkumar reviewed (commented):
More nits.

jaypoulz commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
The OpenShift team raised to us in the 4.2 days that it takes a month for CI to "soak" / stabilize. Since our sprints take 3 weeks, I bumped this to align with sprint boundaries. I also felt that 6 weeks was a better target for internal / external customer feedback gathering.

would like to hear other's thoughts on this.

jaypoulz replied:
I've made some tweaks to this when reformatting onto a Tech + GA format, rather than Dev->Tech->GA

The key with the 6 weeks timer is that it starts when the deliverable is shipped, and the intent is to ensure we have enough time to gather feedback from any prototypes / early adopters and that we're acknowledging and acting upon that feedback.

I can't remember which enhancements proposal I based this on, but I had seen a precedent set for this. That said, I think it might be fair to relax this requirement if there is another way to capture the essence of that commitment to driving quality through the feedback we get. At least insofar as CI, my understanding is that we've found that setting time boundaries ensure we have enough data to actually make determinations about the quality of the release. IDK if that is true for tech preview feedback, but I imagine if we dropped the Tech Preview late in a release cycle, we would want at least some time to absorb feedback before slapping a GA label on.

It's worth noting that platform deliverables have not always been synchronous with GA of an OpenShift release. For example, the ARM developer preview is out now, and that is based on 4.9.

Prashanth684 commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
TBD: figure out how exactly this will happen and if this will happen as part of the terraform provider or the installer. Also "import" would probably be a better choice than "copy" here.

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
For this, are you envisioning a hybird CI approach where we use the existing prow setup and call out to Power VS? And then for the GA, our Power VS IPI CI is entirely on the prow cluster living in Power VS? If so we should discuss with the CI team.

jaypoulz replied:
I've tried to answer this in a new "CI Plan" section. Will want feedback from @deepsm007 and the CI team once I push this draft live.

clnperez reviewed (changes requested):
Some wording updates and a response to your question about criteria. Thanks for adding.

aravindhp commented:
/cc @openshift/openshift-team-cloud

JoelSpeed commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
I think this is similar to how it is done on GCP. We have images in buckets, and then the installer knows a specific string to use for a 4.6/4.7/4.8 cluster respectively. I think this would likely be a preferred approach.

Does adding them to this registry mean they are only available to PowerVS, we were having similar conversations with the IBM VPC team, just wondering if there's a common solution that could be achieved here

Prashanth684 replied:
The idea here is that as part of the RHCOS pipeline, the ova images will be created and pushed to a specific COS bucket in IBMCloud. the replication to multiple regions has to be done manually as there is not automated API to do that. with this PR in the coreos-assembler: https://github.com/coreos/coreos-assembler/pull/2200, the RHCOS pipeline will have the ability to build and push both PowerVS ovas and IBMCloud qcows to COS buckets in IBMCloud. so in that sense RHCOS will support building and pushing images for both IBMCloud and PowerVS. The only thing that needs to be changed is pushing to separate COS buckets for both and not use the same one.

JoelSpeed reviewed (commented):
A couple of general questions:

- Is there any prior art around Machine API style controllers for PowerVS, eg, has an upstream community built a similar controller? Could this be used as a basis?
- Are there similarities in the APIs between PowerVS and IBM VPC, could we make an abstract IBM provider that works with both? (We have this pattern for multiple flavours of Azure)
- Timelines don't seem to be mentioned in this enhancement, could we add some context around the expected timelines
- Have you researched how other provider configurations look for MAPI, to check that you are similar?

mkumatag commented:
> A couple of general questions:
> 
> * Is there any prior art around Machine API style controllers for PowerVS, eg, has an upstream community built a similar controller? Could this be used as a basis?

I have a working code here - https://github.com/openshift-powervs/cluster-api-provider-powervs, though not ready for upstream but most of the things are working fine like mapi, nodeupdate controller. Tested with machineset as well, things look promising.
> * Are there similarities in the APIs between PowerVS and IBM VPC, could we make an abstract IBM provider that works with both? (We have this pattern for multiple flavours of Azure)

Hardly any similarities except the initial authentication, APIs, parameters, image management are completely different.

mkumatag commented:
> > I have a working code here - https://github.com/openshift-powervs/cluster-api-provider-powervs, though not ready for upstream but most of the things are working fine like mapi, nodeupdate controller. Tested with machineset as well, things look promising.
> 
> Given you already have a repo that is owned by Red Hat (I assume this org is managed by Red Hat?), do we actually need a second repository? Even still, should this new repository be a fork rather than a fresh repo, so that there is a link between the two?

The mentioned repo is just a temporary one we created recently for sharing a purpose with a very few histories of commits that aren't needed, it's almost like an initial code. I'm planning to squash all of them and push it to openshift org.

jaypoulz commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
The basic walkthrough attempts to list out the details of the installation that are specific to power vs, where they might differ from other cloud providers.

I've attempted to incorporate some of these notes in the newest version.

jaypoulz commented:
Latest draft is up for review. :smile_cat:

Prashanth684 commented:
/remove-lifecycle stale

jsafrane commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Please add a note about persistent storage (for PVs / PVCs, not for registry nor VMs) and some short roadmap if / how are you going to add it later.

jaypoulz replied:
@Prashanth684 In our next sync, we should enumerate which PS options we will have ready at launch, which are on our roadmap, and what our priorities look like.

Prashanth684 replied:
as far as i am aware they use NFS, but yes we need to have a discussion and involve the IBM folks too

Prashanth684 commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
@mkumatag @clnperez could we add the diagram that you have in your presentation explaining the load balancing topology with a small explanation here?

jaypoulz replied:
Added this under the "Design Details" heading

jaypoulz replied:
We could still probably use an explanation of the load balancing though.

Prashanth684 commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
@clnperez @mkumatag could we also add the high level diagram that was in the presentation explaining the high level data flow and the interaction of different components in the IBMCloud?

jaypoulz replied:
Is this covered by the image under "Design Details"?

Prashanth684 replied:
i was referring to the other diagram you had on your slides (slide 6) which explains the overall interaction of the IBMCloud specific components and the Power VS specific components.

jaypoulz replied:
Awaiting ack to add this

Prashanth684 commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
while this statement is true, the size here (the 16G) refers to the size of the raw disk file inside the ova not the bootimage. just as an fyi

jaypoulz replied:
Added this note as a clarifying statement further in the section.

jaypoulz commented:
/assign JoelSpeed staebler elmiko

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
this should be available via the terraform provider. and we can strikethrough (or delete) 2) below

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
We do need a section to cover DNS though:
"The Power VS provider will use the same DNS solution as the IBM Cloud provider: DNS provided by IBM CIS: https://cloud.ibm.com/catalog/services/internet-services#about. A user will be required to create a DNS domain in CIS before beginning an install and provide it to the installer.

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
this step is done by the user, but I think we should note that all steps onward are done by either TF, bootstrapping, or in-cluster mechanisms such as the MAPI.

clnperez replied:
Another user pre-req we'll need to add is to create a DNS domain in CIS (mentioned above)

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
The new date (10/18) should be added here.

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
this has changed, so the link should now point to https://cloud.ibm.com/catalog/services/internet-services#about or https://cloud.ibm.com/docs/cis?topic=cis-about-ibm-cloud-internet-services-cis (or both?)

JoelSpeed commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Have you tried doing any testing with our E2E suite, might be worth running that against your provider to determine if there are any gaps?

https://github.com/openshift/cluster-api-actuator-pkg

Karthik-K-N replied:
sure, WIll do

jaypoulz replied:
@mkumatag any updates on the actuator e2e results?

mkumatag replied:
we have created a jira task for the same - https://issues.redhat.com/browse/MULTIARCH-1345 and blocked by some other infra issues.

JoelSpeed commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Is there a PowerVS Cloud Controller Manager implementation? Haven't seen it mentioned in here but would expect there to be one

mkumatag replied:
We are trying the following things:
- node controller in the mapi - https://github.com/openshift/cluster-api-provider-powervs/blob/main/pkg/controller/nodeupdate/nodeupdate_controller.go
- service controller for load balancer - reuse the code from IBM cloud(VPC) written, and also use the DNS(CIS) from the ingress operator written.

Hope this can help us fill the gap for now, because we haven't written our own CCM for powervs as of now for the first phase.

Prashanth684 replied:
there is a cloud-provider-ibm (https://github.com/openshift/cloud-provider-ibm). can't that be leveraged ?

mkumatag replied:
> there is a cloud-provider-ibm (https://github.com/openshift/cloud-provider-ibm). can't that be leveraged ?

@Prashanth684 that's what the plan as I mentioned in the above comment except for the node controller which is related to the compute can't be reused the vpc code there.

jaypoulz replied:
Adding a note about this in implementation - section is titled Cloud Controller Manager if someone wants to review

mkumatag replied:
@Karthik-K-N and I can help you reviewing this

mkumatag commented:
/remove-lifecycle stale

staebler commented:
/cc @rna-afk for review from the installer team

rna-afk commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Just curious, how would this be created?

clnperez replied:
we've got this implemented already in Power VS. it's created under the covers when you create a DHCP network in your Service Instance. Honestly we may not have needed to expose this level of detail in the enhancements proposal. And also -- the design changed a bit so taht the VM *is* hidden (and free).

JoelSpeed replied:
Sounds to me like DHCP is now supported, can we get things updated to reflect that?

clnperez replied:
It's only supported for OCP, so, yes, we should rework this section completely with that in mind. I still would rather keep the detail to a minimum if that's ok.

jaypoulz replied:
Yeah since there is dhcp now, I'm just going to remove that detail

JoelSpeed commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
We are actually doing something similar to this on other platforms at the moment, I think this may be the preferred approach in the long term

clnperez replied:
agree this is preferred, but as we don't use the same process for images as ibmcloud, and the power vs folks who handle the image publication process don't have a mechanism for outsiders to provide publicly-available images, this may have to be very long-term. we can open an enhancement request with them (if we haven't already. i need to double-check)

Prashanth684 replied:
now are we talking about "boot" images here or "disk" images ( the OVAs)..i believe when we talk about pre-authorized default images, it is the actual bootimage and not the OVAs. this would be a good distinction to make in the enhancement that what is provided by the RHCOS pipeline today and consumed by the installer is the disk image and not the bootimage.

jaypoulz replied:
I've reworked this section since this isn't under consideration because we don't have a viable way to automate it.

Perhaps we can follow this up with additional enhancements if we can get the Power VS cloud team to add mechanisms for this.

JoelSpeed commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
This may need an update now?

jaypoulz replied:
We've had DHCP for a while now, so I've removed references to it being an open question.

JoelSpeed reviewed (commented):
I've left a few comments in line, there are certainly a few elements of this proposal which are out of date, we should try to get those updated if we can

clnperez commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
@Karthik-K-N 

Is there a specific PR we should point to to make it more clear that the IBM Cloud CCMO supports PowerVS? This commit may be enough? https://github.com/openshift/cluster-cloud-controller-manager-operator/commit/d64dbaa7fc75685ca46d23110fd5b57ca9f4aa38

And would it be accurate to add, "The node objects are still managed by [cluster-api-provider-powervs](https://github.com/openshift/cluster-api-provider-powervs)".

Prashanth684 replied:
this is the PR that adds powervs support to CCM where it instantiates ibm cloud provider assets: https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/129

Karthik-K-N replied:
I have submitted following PRs to use IBM Cloud CCMO and support Power VS
1. https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/129
2. https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/142
3. https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/148

and yes currently the node objects are still managed by cluster-api-provider-powervs:  [Node Controller](https://github.com/openshift/cluster-api-provider-powervs/blob/main/pkg/controller/nodeupdate/nodeupdate_controller.go)

jaypoulz commented:
/hold
For approval until Infra has a chance to finalize review

rna-afk commented:
/lgtm

JoelSpeed commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
This doesn't need updating in the enhancement, but something I wanted to note is that we are trying to align the naming of our providers now to `machine-api-provider-...`, it would be good if we could work to rename this repo before we get it into the product so that it conforms with the other providers for MAPI (this change has been worked on over the last few months)

JoelSpeed reviewed (commented):
/approve

This looks good from my team's perspective.
I can see the installer team has LGTM'd, are there any other teams that we need to get to review this before we merge?

What's the story behind storage for PowerVS? Is this being considered as part of this?

mkumatag commented:
> What's the story behind storage for PowerVS? Is this being considered as part of this?

If it is about csi storage then not part of first phase, we are exploring this for next phase.

jsafrane commented:
> What's the story behind storage for PowerVS? Is this being considered as part of this?

Missing storage (PVs / PVCs) is explicitly noted in the "Persistent Storage" chapter, I find it sufficient. I did not check if there are enough details about storage for VM's root disks and/or registry.

kikisdeliveryservice commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Q: in OCP we are usually talking about Red Hat Enterprise Linux CoreOS (RHCOS) is this what you're referring to or something else?

Prashanth684 replied:
it is the same. the image format in this case is OVA, that's all.

@jaypoulz i think to this point, it might be worth mentioning `Red Hat Enterprise Linux CoreOS` rather than just CoreOS.

kikisdeliveryservice replied:
+1 Prashanth's suggestion above :)

kikisdeliveryservice commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Since this story has a time limit maybe specify a default (3 master-2 worker) cluster?

kikisdeliveryservice reviewed (commented):
A couple of comments and generally, is there an expectation that this would need more from the MCO than the usual changes new providers make to the repo? cc: @jaypoulz

Prashanth684 commented:
> A couple of comments and generally, is there an expectation that this would need more from the MCO than the usual changes new providers make to the repo? cc: @jaypoulz

@kikisdeliveryservice this is the PR for PowerVS support in MCO (https://github.com/openshift/machine-config-operator/pull/2801). The only thing worthy of mention here is that the hostname for the node is configured through the afterburn service which reads the hostname from the metadata in the config drive.

JoelSpeed commented:
/approve cancel

Just to prevent merge until the comments from @kikisdeliveryservice are resolved

jaypoulz commented:
Replaced references from CoreOS to RHCOS where applicable.
Added a note about installations using emptyDir for image registry storage and 120GB disks in their own section, and added Prashanth's note about the MCO afterburn service usage.

jaypoulz commented:
Working with the team to add detail around how ingress interfaces with Direct Link as per request by Network-Edge team

kikisdeliveryservice commented:
From an MCO perspective this seems reasonable especially given that https://github.com/openshift/machine-config-operator/pull/2801 and https://github.com/coreos/afterburn/pull/592 have already merged.

LGTM

cgwalters commented:
Gave this a read through, everything looks sane to me.
/approve

dhellmann commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
I really appreciate that you included the level of detail in this walkthrough. It's very helpful to understand who is doing what, and how.

dhellmann commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
This seems redundant with lines 201 and 215. Is that a copy-and-paste error in duplicating the section? Or an indentation issue with the PR (or the way it's rendered in the diff view)?

jaypoulz replied:
Now with 100% (probably) better indentation :wink:

candita commented:
LGTM, but if possible, could you improve the traffic diagram so it doesn't have Direct Link or properly acknowledges the role of Direct Link?  https://github.com/openshift/enhancements/pull/736/files#diff-bfbf78d36c375706902a827b5aba7f0244f3a25c0edfb52e934332faf63fedd6

candita commented on enhancements/installer/ibm-cloud-for-power-vs-ipi.md:
Something is wrong with the rendering, not sure this fixes it:
```suggestion
3. The user creates a [DNS domain](#dns) in CIS.
```

jaypoulz commented:
Updated formatting and added a note in traffic image about direct link being a transparent GRE tunnel to highlight that thecluster operators don't operate with it directly.

Should be good to merge now:
Acked by:
- installer
- mco
- network-edge
- cloud-infra
- storage
- rhcos

/unhold

JoelSpeed commented:
As far as I can tell, all of the feedback has now been addressed and we have had acks from all of the relevant teams, re-adding my approval
/approve

Prashanth684 commented:
/lgtm
